accord-core/src/main/java/accord/messages/ReadData.java

```diff
@@ -99,12 +99,12 @@ public synchronized void accept(ReadNack reply, Throwable failure)
     {
         if (reply != null)
         {
-            node.reply(replyTo, replyContext, reply);
+            node.reply(replyTo, replyContext, reply, null);
```

Comment:
```suggestion
            node.reply(replyTo, replyContext, reply, failure);
```

you didn't check if failure is non-null, so pass that on to the sink to check

Reply:
Yeah makes sense to have it assert.